docs/replication-retry.md

```diff
@@ -0,0 +1,137 @@
+# Replication Retry
+
+## Description
+
+This feature offers a way for users to monitor and retry failed replication
+operations. It enables users to retrieve a list of failed operations and retry
+specific replication operations.
+
+## Requirements
+
+* The listing of failed operations.
+* The retry of a specific failed operation.
+
+## Design
+
+A RESTful API will expose methods for users to list and retry failed operations.
+See [Explanations](#explanations) for more detail.
+
+## Definition of API
+
+* GET `/_/crr/failed/all`
+
+    This GET request retrieves a listing of all failed operations. This
+    operation is useful if you're interested in whether any replication
+    operations have failed and want the entire listing.
+
+    Response:
+
+    ```sh
+    [{
+        bucket: <bucket>,
+        key: <key>,
+        versionId: <versionId>,
+        site: <site>
+    }]
+    ```
+
+* GET `/_/crr/failed/<bucket>/<key>/<versionId>`
+
+    This GET request retrieves a listing of all failed operations for a specific
+    object version. This operation is useful if you're interested in monitoring
+    the replication status of a specific object.
+
+    Response:
+
+    ```sh
+    [{
+        bucket: <bucket>,
+        key: <key>,
+        versionId: <versionId>,
+        site: <site>
+    }]
+    ```
+
+* POST `/_/crr/failed/retry`
+
+    This POST request retries a failed operation.
+
+    Request Body:
```

Comment:
Does this request only retry specified objects? Maybe also add an option for retrying everything that has failed?

Reply:
I agree that would be a helpful feature, and I will aim to add this route. If it ends up adding too much for the delivery for March 30th, I suppose we can settle on using the response from the `/_/crr/failed/all` as the POST body.